### PR TITLE
implements persistence and transport action for grant/revoke privileges

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

File: users/src/main/java/io/crate/operation/user/PrivilegesRequest.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    public Collection<Privilege> privileges() {
+        return privileges;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (userNames == null) {
```
Comment (andreidan) [inline users/src/main/java/io/crate/operation/user/PrivilegesRequest.java]: should we test for empty usernames/privileges here?

File: users/src/main/java/io/crate/operation/user/PrivilegesResponse.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+public class PrivilegesResponse extends AcknowledgedResponse {
+
+    private long affectedRows;
+
+    PrivilegesResponse() {
+        affectedRows = 1L;
```
Comment (andreidan) [inline users/src/main/java/io/crate/operation/user/PrivilegesResponse.java]: why 1? shouldn't we go with the default 0?

File: users/src/test/java/io/crate/operation/user/TransportPrivilegesActionTest.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    private UsersPrivilegesMetaData setUpMetaData() {
+        Map<String, Set<Privilege>> usersPrivileges = new HashMap<>();
+        for (String userName : USERNAMES) {
+            usersPrivileges.put(userName, new HashSet<>(PRIVILEGES));
+        }
+        return new UsersPrivilegesMetaData(usersPrivileges);
+    }
+
+    @Test
+    public void testApplyPrivilegesFirstTime() throws Exception {
```
Comment (andreidan) [inline users/src/test/java/io/crate/operation/user/TransportPrivilegesActionTest.java]: Please rename this to "testApplyPrivilegesToUserWithoutPrivileges" or something similar as "FirstTime" is too broad

File: users/src/test/java/io/crate/operation/user/TransportPrivilegesActionTest.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+    private static final Privilege GRANT_DQL =
+        new Privilege(Privilege.State.GRANT, Privilege.Type.DQL, Privilege.Clazz.CLUSTER, null, "crate");
+    private static final Privilege GRANT_DML =
+        new Privilege(Privilege.State.GRANT, Privilege.Type.DML, Privilege.Clazz.CLUSTER, null, "crate");
+
+    private static final Set<Privilege> PRIVILEGES = new HashSet<>(Arrays.asList(GRANT_DQL, GRANT_DML));
+    private static final List<String> USERNAMES = Arrays.asList("Ford", "Arthur");
+
+    private UsersPrivilegesMetaData setUpMetaData() {
```
Comment (andreidan) [inline users/src/test/java/io/crate/operation/user/TransportPrivilegesActionTest.java]: Please rename this to "setupDefaultPrivilegesMetadata" for better readability

File: users/src/test/java/io/crate/operation/user/TransportPrivilegesActionTest.java (hunk 3 above)
Comment (seut) [inline users/src/test/java/io/crate/operation/user/TransportPrivilegesActionTest.java]: its testing that not UsersPrivilegesMetaData exists at all. I'll come up with a better name

**Pull request merged**